lib/cloudregister/registerutils.py

```diff
@@ -598,14 +587,56 @@ def set_registry_auth_token(registry_fqdn, username, password):
         list(config_json.get('auths', {}).items()) +  # old content
         list(registry_credentials.items())            # new content
     )
+    logging.info('Writing new credentials')
     updated = write_registry_credentials(config_json)
     return updated
 
 
+# ----------------------------------------------------------------------------
+def set_container_engines_env_vars():
+    """Set the environment variables needed available for
+    the container runtimes to find the config file."""
+    env_vars = [
+        {'REGISTRY_AUTH_FILE': REGISTRY_CREDENTIALS_PATH},
+        {'DOCKER_CONFIG': os.path.dirname(REGISTRY_CREDENTIALS_PATH)}
+    ]
+    export_registry_env_vars = ''.join([
+        '\nexport {}={}\n'.format(engine_var, value)
+        for env_var in env_vars for engine_var, value in env_var.items()
+    ])
+    return update_bashrc(export_registry_env_vars, 'a') and source_env()
+
+
+# ----------------------------------------------------------------------------
+def update_bashrc(content, mode):
+    """Update (add or remove) the env vars for the container engines
+    with the location of the config file to the bashrc local file."""
+    try:
+        with open(BASHRC_LOCAL_PATH, mode) as basrhc_file:
+            basrhc_file.write(content)
```

Comment:
I don't see where the file is getting updated. We simply write the content that is passed in. If the user built their own golden image with a custom `/etc/bash.bashrc.local` file then we just clobbered the users content.

Reply:
We add the two environment vars to the file, whatever the file had

I will modify it accordingly